### PR TITLE
Makes graph nodes rounded

### PR DESCRIPTION
**Pull request by Chaosus**
Users keep requested this feature and @djrm who promised this feature is dissappear. So, I've tried to accept all feedback from #17926 and make a new version. I did not add any new parameters to editor settings, and use Stylebox override to achive this result..

image

image

### CODE REVIEW

File: scene/gui/graph_node.cpp
Comment (toger5) [inline scene/gui/graph_node.cpp]: The theme name should be more descriptive. call it "frame_header"

**Review by toger5 — CHANGES_REQUESTED** [review]
reading this I was like:
WAIT...  this looks too familiar. I could swear I have done that too :)
I checked my local git.
here we go: https://github.com/toger5/godot/commit/a58272e5f93bd1cf4c9c2626de9f707c877b578f
:)

it's a little old though. But still similar. I also had to do (maybe it was just nice...) some calculation for the header hight so the text is always entered and you can make the header hight based on the margins of the style box.

Maybe you can also add the calculation for the header hight. Makes it a lot more flexible for others to use.
(search for `get_header_height`) in the commit I linked.

File: scene/gui/graph_node.cpp
Comment (Chaosus) [reply scene/gui/graph_node.cpp]: Wow, your version looks more clear than mine. Why you dont PR this ?

**Comment by Chaosus** [comment]
@toger5 Wow, your version looks much clear than mine.. Why you didnt PR it ?

**Comment by Chaosus** [comment]
>  also had to do (maybe it was just nice...) some calculation for the header hight so the text is always entered and you can make the header hight based on the margins of the style box.

I just leave it as default 24 * EDSCALE 
Why need to change it ?

**Comment by toger5** [comment]
@Chaosus good question... I really don't know. Maybe there is a little bit missing or I had an additional idea which I really wanted to add to it. no-one knows... it's 10 month ago ;)

I'm really thankful you brought the whole idea back to live though.

**Comment by toger5** [comment]
I think you should copy the header hight and the offset stuff. you used a second dictionary for the header which is nice. it makes it easier to read than my "string combination thing".
It's funny that they are so similar ;)
I think I didn't even change the style though.it looks the same but allows for more advanced theme options.

**Comment by toger5** [comment]
do you want me to merge and rebase on current master? so you have an easier time comparing?

**Comment by djrm** [comment]
I still think this is not a good time for this, all the current theme is square (like it or not), i tried to add rounded corners to the whole theme, however it did not look that good because the 2d antialias 'sucks' (as pointed out by reduz), so this is on hold, i still plan to do some theme work for the next version.

**Comment by Chaosus** [comment]
@djrm Ah, thankfully you are alive !
@toger5 I dont know really, reduz said his "no" in irc and probably @djrm have the correct solution
Then I will close this

**Comment by groud** [comment]
I like the round theme for graph nodes. It makes them feel more like they can be moved around, more "floaty" that the squared blocks. If we keep that only for nodes in graphs I do not think it is a problem.

Also, I think it's prettier ^^

**Comment by Chaosus** [comment]
@groud Should I reopen it ? I really dont know..

**Comment by toger5** [comment]
There is aa for styleboxes, and it rooks good enough.
I would not consider this as an arguement against the feature. The way it is implemented also should make it look okay when you zoom in.
Adding two styleboxes to graph nodes is with bo quedtion an improovement, also for ppl creating games. Otherwise graph node styles are quiet limited.

My proposal:
@Chaosus reopens and adds proper the auto header scaling + auto close button positioning. And you could split it knto two commits. One for the updated look and one for the additional graph node faerures.

**Comment by toger5** [comment]
Than we merge really quick and decide before 3.1 if the updated style gets reverted or not.

**Comment by toger5** [comment]
I also could do the rebase on my version.. it's more or less the same.(except the auto header height cacluation)

**Comment by Chaosus** [comment]
@toger5 I have some troubles with your version. You can add this auto height calculation after, if this PR is ever been merged, so I leave it opened as it is now.

File: scene/gui/graph_node.cpp
Comment (akien-mga) [inline scene/gui/graph_node.cpp]: Keep in mind that `scene/` stuff is used in games... So you can't use editor stuff here. And you should avoid breaking compat if possible.

Comment (Chaosus) [reply scene/gui/graph_node.cpp]: Fixed !

**Pull request closed without merging**
